### PR TITLE
transactionally resume time partitioned table schema changes

### PR DESCRIPTION
**Pull request by ldennis**
.

### CODE REVIEW

**Comment by dorinhogea** [comment]
I am seeing some issues resuming an interrupted tpt schema change:
[ERROR] bdb_fetch_int_ll: dtafile=15 out of range
[ERROR] bdb_fetch_next_dtastripe_record: bdb_fetch_int rc -1 bdberr 14
[ERROR] *ERROR* bdb_fetch_next_dtastripe_record unhandled rcode 14
[ERROR] Schema change error: error -1 reading database records
[ERROR] Schema change error: conversion failed after 0 records, while working on stripe 6 with 0 retries
[ERROR] Schema change error: Stoping work on stripe 7 because the thread for stripe 6 failed
[ERROR] Schema change error: conversion failed after 0 records, while working on stripe 7 with 0 retries
[ERROR] Schema change error: convert_all_records failed
[ERROR] Schema change error:   > stripe  0 was at 0xffffffffffffffff
[ERROR] Schema change error:   > stripe  1 was at 0xffffffffffffffff
[ERROR] Schema change error:   > stripe  2 was at 0xffffffffffffffff
[ERROR] Schema change error:   > stripe  3 was at 0xffffffffffffffff
[ERROR] Schema change error:   > stripe  4 was at 0xffffffffffffffff
[ERROR] Schema change error:   > stripe  5 was at 0xffffffffffffffff
[ERROR] Schema change error:   > stripe  6 was at 0xffffffff00000000
[ERROR] Schema change error:   > stripe  7 was at 0x0000000000000000
[ERROR] Schema change error: removing temp table for <$1_E5EE49E6>
[ERROR] >>> SCHEMA CHANGE ERROR: TABLE $1_E5EE49E6, RC 2
In another context, I get this crash while doing a tpt alter:
Thread 44 "comdb2" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffea651700 (LWP 188806)]
0x0000000000628202 in start_schema_change_tran (iq=0x7fffeb11aa20, trans=0x0) at /bb/bigstorn/systems/dhogea/comdb2/OPEN/ldennis/schemachange/schemachange.c:205
205             if (!s->db->doing_upgrade || s->fulluprecs || s->partialuprecs) {
(gdb) print s->db
$31 = (struct dbtable *) 0x0
(gdb) print s->table    
$32 = "$1_E5EE49E6", '\000' <repeats 20 times>

**Comment by ldennis** [comment]
`doing_upgrade` is not currently used I think. iirc, upgrade is not longer needed?

File: schemachange/sc_logic.c [hunk 1]
```diff
@@ -623,11 +623,32 @@ void *sc_resuming_watchdog(void *p)
     return NULL;
 }
 
+struct timepart_sc_resuming {
+    char *viewname;
+    struct schema_change_type *s;
+};
+
```
[the remark this answers is not in the document]
Comment (ldennis) [reply schemachange/sc_logic.c]: Yes, transaction that has multiple sc will require bplog resubmission to resume. That part is not perfect at this moment.. I will work on that later.

File: schemachange/sc_logic.c [hunk 2]
```diff
@@ -680,15 +706,14 @@ int resume_schema_change(void)
...
 
             free(packed_sc_data);
 
             /* Give operators a chance to prevent a schema change from resuming.
              */
-            char *abort_filename =
+            abort_filename =
                 comdb2_location("marker", "%s.scabort", thedb->envname);
             if (access(abort_filename, F_OK) == 0) {
                 rc = bdb_set_in_schema_change(NULL, thedb->dbs[i]->tablename,
```
[the remark this answers is not in the document]
Comment (ldennis) [reply schemachange/sc_logic.c]: sure.

**Pull request merged**